tests/testflows/rbac/tests/syntax/alter_quota.py

```diff
@@ -0,0 +1,206 @@
+
+from testflows.core import *
+
+from rbac.requirements import *
+import rbac.tests.errors as errors
+
+@TestFeature
+@Name("alter quota")
+@Args(format_description=False)
+def feature(self, node="clickhouse1"):
+    """Check alter quota query syntax.
+
+    ```sql
+    ALTER QUOTA [IF EXISTS] name [ON CLUSTER cluster_name]
+    [RENAME TO new_name]
+    [KEYED BY {'none' | 'user name' | 'ip address' | 'client key' | 'client key or user name' | 'client key or ip address'}]
+    [FOR [RANDOMIZED] INTERVAL number {SECOND | MINUTE | HOUR | DAY | MONTH}
+        {MAX { {QUERIES | ERRORS | RESULT ROWS | RESULT BYTES | READ ROWS | READ BYTES | EXECUTION TIME} = number } [,...] |
+        NO LIMITS | TRACKING ONLY} [,...]]
+    [TO {role [,...] | ALL | ALL EXCEPT role [,...]}]
+    ```
+    """
+    node = self.context.cluster.node(node)
+
+    def cleanup_quota(quota):
+        with Given(f"I ensure that quota {quota} does not exist"):
+            node.query(f"DROP QUOTA IF EXISTS {quota}")
+
+    try:
+        with Given("I have a quota, a user, and a role"):
+            node.query(f"CREATE QUOTA quota0")
+            node.query(f"CREATE USER user0")
+            node.query(f"CREATE ROLE role0")
+
+        with Scenario("I alter quota with no options", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter("1.0")]):
+            with When("I alter quota"):
+                node.query("ALTER QUOTA quota0")
+
+        with Scenario("I alter quota that does not exist, throws an exception", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter("1.0")]):
+            quota = "quota1"
+            cleanup_quota(quota)
+            with When(f"I alter quota {quota}, which does not exist"):
+                exitcode, message = errors.quota_not_found_in_disk(name=quota)
+                node.query(f"ALTER QUOTA {quota}", exitcode=exitcode, message=message)
+            del quota
+
+        with Scenario("I alter quota with if exists, quota does exist", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter_IfExists("1.0")]):
+            node.query("ALTER QUOTA IF EXISTS quota0")
+
+        with Scenario("I alter quota with if exists, quota does not exist", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter_IfExists("1.0")]):
+            quota = "quota1"
+            cleanup_quota(quota)
+            with When(f"I alter quota {quota}, which does not exist, with IF EXISTS"):
+                node.query(f"ALTER QUOTA IF EXISTS {quota}")
+            del quota
+
+        with Scenario("I alter quota using rename, target available", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter_Rename("1.0")]):
+            node.query("ALTER QUOTA quota0 RENAME TO quota0")
+
+        with Scenario("I alter quota using rename, target unavailable", flags=TE, requirements=[
+                RQ_SRS_006_RBAC_Quota_Alter_Rename("1.0")]):
+            new_quota = "quota1"
+
+            try:
```

Comment:
What's the difference between `Given` and `When`? It looks like in both cases you do something and you check something.

Reply:
You can find more information in the TestFlows handbook. Here are a few links: https://testflows.com/handbook/#Given, and https://testflows.com/handbook/#When.